e2e/playground.py

```diff
@@ -0,0 +1,2135 @@
+        id: str,
+    ) -> None:
+        super().__init__(
+            page,
+            id=id,
+            loc=f"button#{id}.action-button.shiny-bound-input",
+        )
+
+
+class InputActionLink(_InputActionBase):
+    # label: TagChildArg,
+    # icon: TagChildArg = None,
+
+    def __init__(
+        self,
+        page: Page,
+        id: str,
+    ) -> None:
+        super().__init__(
+            page,
+            id=id,
+            loc=f"a#{id}.action-button.shiny-bound-input",
+        )
+
+
+# * click:
+#     * input_checkbox_group
+#     * input_radio_buttons
+
+
+class InputCheckboxBase(
+    _WidthContainer,
+    _InputWithLabel,
+):
+    # label: TagChildArg
+    # value: bool = False
+    # width: Optional[str] = None
+    def __init__(self, page: Page, id: str, loc: InitLocator) -> None:
+        super().__init__(
+            page,
+            id=id,
+            loc=loc,
+        )
+
+    def set(
+        self, value: bool, *, timeout: Timeout = None, **kwargs: typing.Any
+    ) -> None:
+        self.loc.set_checked(value, timeout=timeout, **kwargs)
+
+    def toggle(self, *, timeout: Timeout = None, **kwargs: typing.Any) -> None:
+        self.loc.click(timeout=timeout, **kwargs)
+
+    def value(self, *, timeout: Timeout = None) -> bool:
+        return self.loc.is_checked(timeout=timeout)
+
+    def expect_to_be_checked(self, value: bool, *, timeout: Timeout = None) -> None:
+        if value:
+            self.expect.to_be_checked(timeout=timeout)
+        else:
+            self.expect.not_to_be_checked(timeout=timeout)
+
+
+class InputCheckbox(InputCheckboxBase):
+    def __init__(
+        self,
+        page: Page,
+        id: str,
+    ) -> None:
+        super().__init__(
+            page,
+            id=id,
+            loc=f"input#{id}[type=checkbox].shiny-bound-input",
+        )
+
+
+class InputSwitch(InputCheckboxBase):
+    def __init__(
+        self,
+        page: Page,
+        id: str,
+    ) -> None:
+        super().__init__(
+            page,
+            id=id,
+            loc=f"input#{id}[type=checkbox].shiny-bound-input",
+        )
+
+
+class _RadioButtonCheckboxGroupBase(_InputWithLabel):
+    loc_choice_labels: Locator
+
+    def expect_choice_labels(
+        self,
+        labels: typing.Union[PatternOrStr, typing.List[PatternOrStr]],
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        playwright_expect(self.loc_choice_labels).to_have_text(
+            labels,
+            timeout=timeout,
+        )
+
+    def expect_inline(self, inline: bool, *, timeout: Timeout = None) -> None:
+        if inline:
+            playwright_expect(self.loc_container).to_have_class(
+                "shiny-input-container-inline", timeout=timeout
+            )
+        else:
+            playwright_expect(self.loc_container).not_to_have_class(
+                "shiny-input-container-inline", timeout=timeout
+            )
+
+    def _expect_locator_values_in_list(
```

Comment:
This could be a standalone utility function

Reply:
Even though I requested this, I'm keeping it here as it only pertains to this base class